src/main/java/programminglife/gui/controller/GuiCreateBookmarkController.java

```diff
@@ -0,0 +1,134 @@
+package programminglife.gui.controller;
+
+
+import javafx.fxml.FXML;
+import javafx.scene.control.Alert;
+import javafx.scene.control.Button;
+import javafx.scene.control.TextArea;
+import javafx.scene.control.TextField;
+import javafx.stage.Stage;
+import programminglife.controller.BookmarkController;
+
+/**
+ * Class for the GuiCreateBookmarkController. This class handles the FXML file that comes with it.
+ */
+public class GuiCreateBookmarkController {
+
+    private String graphName;
+    private GraphController graphController;
+
+    @FXML private Button btnOk;
+    @FXML private Button btnCancel;
+    @FXML private TextField txtBookmarkName;
+    @FXML private TextField txtId;
+    @FXML private TextField txtRadius;
+    @FXML private TextArea txtDescription;
+
+    /**
+     * Initialize method for BookmarkController.
+     */
+    @FXML
+    @SuppressWarnings("unused")
+    public void initialize() {
+        initButtons();
+    }
+
+    /**
+     * Initializes the buttons in the window.
+     */
+    private void initButtons() {
+        btnOk.setOnAction(event -> {
+            if (txtBookmarkName.getText().contains(" ")) {
+                nameSpaceAlert();
+            } else if (txtBookmarkName.getText().matches(".*\\d+.*")) {
+                nameIntAlert();
+            } else if (!txtId.getText().matches("^[1-9]\\d*$")) {
+                intAlert("Center Node");
+            } else if (Integer.parseInt(txtId.getText()) > graphController.getGraph().size()) {
+                sizeAlert(graphController.getGraph().size());
+            } else if (!txtRadius.getText().matches("^[1-9]\\d*$")) {
+               intAlert("radius");
+            } else {
+                BookmarkController.storeBookmark(graphName, txtBookmarkName.getText(), txtDescription.getText(),
+                        Integer.parseInt(txtId.getText()), Integer.parseInt(txtRadius.getText()));
+                Stage s = (Stage) btnOk.getScene().getWindow();
+                s.close();
+            }
+        });
+        btnCancel.setOnAction(event -> {
+            Stage s = (Stage) btnCancel.getScene().getWindow();
+            s.close();
+        });
+    }
+
+    /**
+     * Alert for title if numbers are in name.
+     */
+    private void nameIntAlert() {
```

Comment:
I will group these alert methods into one method with title and message parameters.